src/rgw/rgw_bucket.cc

```diff
@@ -512,8 +512,16 @@ int rgw_remove_object(RGWRados *store, RGWBucketInfo& bucket_info, rgw_bucket& b
   }
 
   rgw_obj obj(bucket, key);
-
-  return store->delete_obj(rctx, bucket_info, obj, bucket_info.versioning_status());
+  int r = store->delete_obj(rctx, bucket_info, obj, bucket_info.versioning_status());
+  if (r == -ENOENT) {
+    if (obj.key.instance == "null") {
```

Comment:
@zhangsw Are we sure at this point that `string instance;` of `struct rgw_obj_key` would be null not ""?

Reply:
Yes, we have changed it to "null" above.